pkg/dashboard/server/auth.go

```diff
@@ -66,7 +68,37 @@ func auth(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	http.RedirectHandler("/", http.StatusSeeOther).ServeHTTP(w, r)
+	redirect, err := redirectPath(r)
+	if err != nil {
+		log.Errorf("cannot get redirect path: %s", err)
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	http.RedirectHandler(redirect, http.StatusSeeOther).ServeHTTP(w, r)
+}
+
+func redirectPath(r *http.Request) (string, error) {
+	redirect := "/"
+	err := r.ParseForm()
+	if err != nil {
+		return "", fmt.Errorf("cannot parse form: %s", err)
+	}
+
+	state := r.Form.Get("state")
```

Comment:
Please write a unit test for this, so the URL path is better communicated for our future selves.